src/sinks/datadog/events/service.rs

```diff
@@ -41,13 +39,13 @@ impl DriverResponse for DatadogEventsResponse {
     }
 
     fn bytes_sent(&self) -> Option<(usize, &str)> {
-        Some((self.raw_byte_size, &self.protocol))
+        // HttpBatchService emits EndpointBytesSend
+        None
```

Comment:
I see there are a few other sinks using `HttpBatchService` . I think the easiest approach to see all the ramifications and not miss one, is to make the change and see what happens in CI.

Reply:
However, :thinking: probably still need to inspect it further, as not all sinks are asserting compliance yet.

Reply:
Well from the CI run at least, these sinks seem to rely on `HttpBatchService` emitting `EndpointBytesSent` :

```
splunk_hec
influxdb
humio
gcp
elasticsearch
clickhouse
axiom
apex
```

Reply:
Sinks from that list which are "new style" and are currently NOT implementing `bytes_sent()` but could :
```
elasticsearch,  [axiom]
splunk_hec, [humio]
```

That leaves these sinks as needing special treatment, and they all seem to utilize `BatchedHttpSink`
```
apex
influxdb
clickhouse
gcp_stackdriver_logs
```

Reply:
`BatchedHttpSink` uses `HttpBatchService`.

Ok, so it sounds like the options are :

1. Modify `HttpBatchService` to have a flag to gate whether to emit `EndpointBytesSent` , and new style sinks set the flag to true.
2. Modify the sinks using `BatchedHttpSink` (or modify `BatchedHttpSink` itself) to emit `EndpointBytesSent`.

I would think option 2 might be preferred since if `BatchedHttpSink will eventually go away, we would not have to come back and remove the flag to `HttpBatchService` when all sinks are in the new style.

Thoughts ?

Reply:
As discussed, I've opened a new issue to track this: https://github.com/vectordotdev/vector/issues/14340

Note, I've got some changes on branch as progress towards this.